src/text-filter/search-results.tsx

```diff
@@ -15,12 +15,52 @@ interface SearchResultsProps {
   children: string;
 }
 
-export function SearchResults({ id, children }: SearchResultsProps) {
-  return (
-    <span className={styles.results}>
-      <InternalLiveRegion delay={LIVE_REGION_DELAY} tagName="span">
-        <span id={id}>{children}</span>
-      </InternalLiveRegion>
-    </span>
-  );
-}
+export const SearchResults = React.forwardRef(
+  ({ id, children }: SearchResultsProps, ref?: React.Ref<InternalLiveRegionRef>) => {
+    const [announcementText, setAnnouncementText] = useState('');
+    const isComponentMounted = useRef(false);
+    const liveRegionRef = useRef<InternalLiveRegionRef>(null);
+
+    useEffect(() => {
+      isComponentMounted.current = true;
+      return () => {
+        isComponentMounted.current = false;
+      };
+    }, []);
+
+    // Consolidate multiple re-announce calls into a single re-announcement.
+    const announceDebounced = useDebounceCallback(() => {
+      if (!isComponentMounted.current) {
+        return;
+      }
+      setAnnouncementText(children);
+      liveRegionRef.current?.reannounce();
+    }, LIVE_REGION_DELAY);
+
+    useImperativeHandle(
+      ref,
+      () => ({
+        reannounce: () => {
+          announceDebounced();
+        },
+      }),
+      [announceDebounced]
+    );
+
+    useEffect(() => {
+      setAnnouncementText('');
+      announceDebounced();
```

Comment:
Keeping track of the text to announce in separate state. This is to control the 2 second delay on the initial render and the `reannounce` call in one central place.